crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -230,7 +230,9 @@ impl Inner {
                                 "automatic check error: order_fees missing"
                             );
                         } else {
-                            let settlement_fee = order_fees[&domain::OrderUid(fee.0 .0)];
+                            let settlement_fee = order_fees[&domain::OrderUid(fee.0 .0)]
+                                .clone()
```

Comment:
To avoid cloning the whole thing since you only want one field.
```suggestion
                                .as_ref()
```

Reply:
added